lib/src/scss-shared/_nx-form-validation.scss

```diff
@@ -5,20 +5,56 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 @import 'nx-variables';
+@import 'nx-container-helpers';
 
 @mixin valid() {
-  border-left-color: $nx-valid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-valid-color;
+  }
+
+  .nx-text-input__valid-message {
+    display: inline-block;
+  }
 }
 
 @mixin invalid() {
-  border-color: $nx-invalid-color;
-  border-left-style: solid;
+  .nx-text-input__input:not(:disabled) {
+    border-color: $nx-invalid-color;
+  }
+
+  .nx-text-input__invalid-message {
+    display: block;
+  }
 }
 
 @mixin pristine() {
-  border-color: $nx-grey-4;
-  border-left-color: $nx-valid-color;
-  border-left-style: double;
-  color: $nx-text-color-light;
+  .nx-text-input__input:not(:disabled) {
+    border-color: #d1d6f0;
+    color: $nx-text-color-light;
+  }
+}
+
+.nx-text-input__valid-message, .nx-text-input__invalid-message {
+  display: none;
+}
+
+.nx-icon--validation-check {
```

Comment:
Shouldn't this block be in the dedicated scss file for that icon?

Reply:
> Shouldn't this block be in the dedicated scss file for that icon?

Moved here: https://github.com/sonatype/sonatype-react-shared-components/pull/57/commits/d0e175eb0abc97c26d1ce9f69d6d05aeecd1a92e